corehq/apps/users/util.py

```diff
@@ -51,6 +51,7 @@ def normalize_username(username, domain=None):
     """
     from django.core.validators import validate_email
 
+    username = str(username)
```

Comment:
how confident are you that this is always the correct thing to do? (as opposed to e.g. a more surgical check like only converting numbers)

Reply:
I think this is the correct action though with @mkangia :
```
if not username:
    return username
username = str(username)
```

Reply:
This is coming from the error based on which it looks `re.sub` only expect `string or bytes-like object` So it looks safe to convert usernames to string.

Reply:
Actually since this raises a validation error for invalid emails I think this may be more correct
```python
if not username:
    raise ValidationError
username = str(username)
```

Reply:
ok making that

Reply:
@snopoke 
Just curious,
isn't `ValidationError` more applicable to models/forms and this being a generic method which has no knowledge of them, not raise this exception?

Reply:
the function already raises ValidationError if the email is invalid so keeping it consistent